acme/acme.go

```diff
@@ -257,6 +259,86 @@ func (c *Client) RevokeCert(ctx context.Context, key crypto.Signer, cert []byte,
 	return c.revokeCertRFC(ctx, key, cert, reason)
 }
 
+// FetchRenewalInfo retrieves the RenewalInfo from Directory.RenewalInfoURL.
+func (c *Client) FetchRenewalInfo(ctx context.Context, leaf, issuer []byte) (*RenewalInfo, error) {
+	if _, err := c.Discover(ctx); err != nil {
+		return nil, err
+	}
+
+	parsedLeaf, err := x509.ParseCertificate(leaf)
+	if err != nil {
+		return nil, err
+	}
+	parsedIssuer, err := x509.ParseCertificate(issuer)
+	if err != nil {
+		return nil, err
+	}
+
+	renewalURL, err := c.getRenewalURL(parsedLeaf, parsedIssuer)
+	if err != nil {
+		return nil, err
```

Comment:
This code is all in the style of this package already, so maybe you'll want to leave it like this, but one challenge I face when reading code like this is if asn1.Unmarshal fails in getRenewalURL, then it will bubble up through getRenewalURL and through FetchRenewalInfo without any annotation, and it can be extremely difficult to figure out where the error came from. So some exported APIs call for prefixing errors. Where exactly is up to you, it's a bit of a judgement call and it's not obvious to me here.